methods.rb

```diff
@@ -0,0 +1,96 @@
+module Enumerable
+  def my_each
+    for i in self
+      yield(i) if block_given?
+    end
+  end
+
+  def my_each_with_index
+    i = 0
+    for j in self
+      yield(j, i) if block_given?
+      i += 1
+    end
+  end
+
+  def my_select
+    result = []
+    self.my_each do |i|
+      result.<<(i) if yield(i)
+    end
+    result
+  end
+
+  def my_all?
+    return true if !block_given?
+    if self.is_a? Hash
+      self.my_each { |m, n| return false if !yield(m, n) } 
+    else
+      self.my_each { |i| return false if !yield(i) }
+    end
+    true
+  end
+
+  def my_any?
+    return true if !block_given?
+    if self.is_a? Hash
+      self.my_each { |m, n| return true if yield(m, n) }
+    else
+      self.my_each { |i| return true if yield(i) }
+    end
+    false
+  end
+
+  def my_none?
+    return false unless block_given?
+    if self.class == Hash
+      self.my_each do |m, n|
+        return false if yield(m, n)
+      end
+    else
+      self.my_each do |i|
+        return false if yield(i)
+      end
+    end
+    true
+  end
+
+  def my_count?
```

Comment:
This method does not return a boolean. It should be named `my_count`. It's not asking a question it's simply giving an answer.

Additional Requirements for this method:
it should:

- return the count of the number of items in enum that are equal to item if an argument is given